internal/docker/deployer.go

```diff
@@ -148,6 +156,142 @@ func (d *Deployer) Destroy(dep *Deployment, printServerLogs bool) {
 	}
 }
 
+func deployImage(
+	docker *client.Client, imageID string, csPort int, containerName, pkgNamespace, blueprintName, hsName string,
+	asIDToRegistrationMap map[string]string, contextStr, networkID string, spawnHSTimeout time.Duration,
+) (*HomeserverDeployment, error) {
+	ctx := context.Background()
+	var extraHosts []string
+	var mounts []mount.Mount
+	var err error
+
+	if runtime.GOOS == "linux" {
+		// By default docker for linux does not expose this, so do it now.
+		// When https://github.com/moby/moby/pull/40007 lands in Docker 20, we should
+		// change this to be  `host.docker.internal:host-gateway`
+		extraHosts = []string{HostnameRunningComplement + ":172.17.0.1"}
+	}
+
+	toMount := []Volume{
+		&VolumeCA{}, &VolumeAppService{},
```

Comment:
The other difference here is that it always creates the `VolumeCA` mount regardless of whether the `COMPLEMENT_CA` environment variable is set